pkg/controller/node/nodecontroller.go

```diff
@@ -746,6 +748,38 @@ func (nc *NodeController) deletePods(nodeName string) (bool, error) {
 	return remaining, nil
 }
 
+// update ready status of all pods running on given node from master
+// return true if success
+func (nc *NodeController) updatePods(nodeName string, readyStatus api.ConditionStatus) (bool, error) {
+	glog.Infof("Update Ready Status of pods on node [%v]", nodeName)
+
+	pods, err := nc.kubeClient.Pods(api.NamespaceAll).List(labels.Everything(), fields.OneTermEqualSelector(client.PodHost, nodeName))
+	if err != nil {
+		return false, err
+	}
+
+	for _, pod := range pods.Items {
+		// Defensive check, also needed for tests.
+		if pod.Spec.NodeName != nodeName {
+			continue
+		}
+
+		for i, cond := range pod.Status.Conditions {
+			if cond.Type == api.PodReady {
+				pod.Status.Conditions[i].Status = readyStatus
+				pod, err := nc.kubeClient.Pods(pod.Namespace).UpdateStatus(&pod)
+				if err != nil {
+					glog.Warningf("Failed to updated status for pod %q: %v", kubeletutil.FormatPodName(pod), err)
+					return false, err
```

Comment:
can this return an aggregated error instead?

rather than breaking out here and returning false directly, can this just ```continue``` so each pod is visited?

right now it looks like if one pod update fails, the others are not updated which is not ideal.